services/obs-operator/pkg/utils/duration_test.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"testing"
+)
+
+func TestPromDurationToISO8601(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{"2w", "P2W"},
+		{"1ms", "PT0.001S"},
+		{"1000ms", "PT1S"},
+		{"1y2w3d4h5m6s", "P1Y2W3DT4H5M6S"},
+		{"1y2w3d4h5m6s7ms", "P1Y2W3DT4H5M6.007S"},
```

Comment:
Regarding this: https://github.com/Altinn/altinn-platform/pull/1018/files#r1812522355

Also another test case may need to get added here:

```
{"0", "PT0S"},
```

Reply:
Go ahead, you can just add them. See what happens in those tests.